### PR TITLE
Add ContainerProviderExtension

### LINKED ISSUE #2041: [Enhancement] Add ContainerProvider for XAML in WPF
## Summary

It enables the same function as #1347 in WPF.

## API Changes

Add only one class.

```cs
public class ContainerProviderExtension : System.Windows.Markup.MarkupExtension 
{
  public ContainerProviderExtension(Type t) => Type = t;

  private static readonly IContainerProvider _containerProvider
    => (Application.Current as PrismApplicationBase)?.Container;

  public Type Type { get; set; }
  public String Name { get; set; }

  public override object ProvideValue(IServiceProvider _)
    => _containerProvider.Resolve(Type, Name);
}
```

e.g.

```xml
<Window.DataContext>
  <prism:ContainerProvider Type="local:MyVM" />
</Window.DataContext>
<TextBlock Text="{Binding Path=Foo,
                          Converter={prism:ContainerProvider {x:Type local:MyConverter}}}" />
```

### PR DESCRIPTION
**Pull request by gayaK**
﻿## Description of Change

Fixes #2041

### API Changes

Added:

- XAML markup extension: `{prism:ContainerProvider Type={x:Type MyType}, Name=TargetName}`

### PR Checklist

- [x] Has tests (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

**Comment by brianlagunas** [comment]
Could you please provide unit tests for this feature?

File: src/Wpf/Prism.Wpf/Ioc/ContainerProviderExtension.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    /// Usage as XML element:
+    /// <![CDATA[
+    ///   <Window>
+    ///     <Window.DataContext>
+    ///       <prism:ContainerProvider Type="{x:Type local:MyViewModel}" />
+    ///     </Window.DataContext>
+    ///   </Window>
+    /// ]]>
+    /// </para>
+    /// </example>
```
Comment (dansiegel) [inline src/Wpf/Prism.Wpf/Ioc/ContainerProviderExtension.cs:29]: if you see Microsoft Docs they show proper use of the example block inside of the summary block. https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments#code

**Comment by gayaK** [comment]
@brianlagunas 
Yes, please wait several days.

**Comment by gayaK** [comment]
Sorry. I'm fixing CI errors.

**Comment by brianlagunas** [comment]
Thank you so much for the PR. Sorry it took so long for me to get to it.

**Pull request merged**